src/uct/ib/rdmacm/rdmacm_cm_ep.c

```diff
@@ -149,58 +149,33 @@ ucs_status_t uct_rdmacm_cm_ep_conn_notify(uct_ep_h ep)
     return cep->status;
 }
 
-static ucs_status_t
-uct_rdamcm_cm_ep_is_reserved_qpn_supported(uct_rdmacm_cm_ep_t *cep, bool *supported)
-{
-    struct uct_rdmacm_cm_reserved_qpn_context *ctx;
-    ucs_status_t status;
-
-    status = uct_rdmacm_cm_get_reserved_qpn_context(uct_rdmacm_cm_ep_get_cm(cep),
-                                                    cep->id->verbs, &ctx);
-    if (status == UCS_OK) {
-        *supported = ctx->support;
-    }
-
-    return status;
-}
-
 /**
  * Allocate a reserved QPN either from the last FW object allocated,
  * or by allocating a new one. When find a free QPN in an object, it
  * always starts from last allocation position, to make sure the QPN
  * always move forward to prevent stale QPN.
  */
-static ucs_status_t uct_rdamcm_cm_ep_create_reserved_qpn(uct_rdmacm_cm_ep_t *cep)
+static ucs_status_t uct_rdamcm_cm_ep_create_reserved_qpn(uct_rdmacm_cm_ep_t *cep,
+                                                         struct uct_rdmacm_cm_qp_context *ctx)
 {
-    struct uct_rdmacm_cm_reserved_qpn_context *ctx;
     struct uct_rdmacm_cm_reserved_qpn_blk *blk;
     uint32_t qpns_per_obj;
-    ucs_status_t status;
-
-    status = uct_rdmacm_cm_get_reserved_qpn_context(uct_rdmacm_cm_ep_get_cm(cep),
-                                                    cep->id->verbs, &ctx);
-    if (status != UCS_OK) {
-        return status;
-    }
+    ucs_status_t status = UCS_OK;
 
-    if (!ctx->support) {
-        return UCS_ERR_UNSUPPORTED;
-    }
-
-    qpns_per_obj = 1 << ctx->log_reserved_qpn_granularity;
+    qpns_per_obj = 1 << ctx->reserved_qpn_ctx.log_reserved_qpn_granularity;
```

Comment:
maybe store calculated once `reserved_qpn_granularity` instead of `log_reserved_qpn_granularity`?

Reply:
@evgeny-leksikov Fixed.